src/gradio_ui.py

```diff
@@ -100,6 +100,46 @@ def _regenerate(
         def _undo_last_exchange(model_history, gradio_history):
             '''Undoes the last exchange (message pair).'''
             return model_history[:-2], gradio_history[:-1], gradio_history[:-1]
+            
+        def _save_chat_history(gradio_history, *char_setting_states):
+            '''Saves the current chat history to a .json file.'''
+            char_name = char_setting_states[0]
+            with open(f"{char_name}_conversation.json", "w") as f:
+                f.write(json.dumps({"chat": gradio_history}))
+            return f"{char_name}_conversation.json"
+            
+                
+        def _load_chat_history(file_obj, *char_setting_states):
+            '''Loads up a chat history from a .json file.'''
+            file_data = json.loads(file_obj.decode('utf-8'))
+            gradio_history = file_data["chat"]
+            char_name = char_setting_states[0]
```

Comment:
Does this function need to take in the entirety of `char_setting_states` if it only uses `char_name`?

Reply:
I feel that having `char_setting_states` in those functions would make the code rather consistent with how the rest of functions there have `char_setting_states` as input rather than feeding in `char_setting_states[0]` to them. But I also just realized that yeah, functionality before code consistency kek

Reply:
Fixed in new commit